src/main/java/com/cloudbees/jenkins/support/AnonymizedItems.java

```diff
@@ -0,0 +1,148 @@
+import hudson.Extension;
+import hudson.model.ManagementLink;
+import hudson.util.FormApply;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import javax.annotation.CheckForNull;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Action for viewing anonymized items mapping.
+ */
+@Extension
+public class AnonymizedItems extends ManagementLink {
+    private SupportPlugin plugin;
+    private SupportPlugin.AnonymizationSettings settings;
+
+    public AnonymizedItems() {
+        plugin = SupportPlugin.getInstance();
+        settings = plugin.getAnonymizationSettings();
+    }
+
+    @CheckForNull
+    @Override
+    public String getIconFileName() {
+        return "/plugin/support-core/images/24x24/support.png";
+    }
+
+    @CheckForNull
+    @Override
+    public String getDisplayName() {
+        return Messages.AnonymizedItems_DisplayName();
+    }
+
+    @CheckForNull
+    @Override
+    public String getUrlName() {
+        return "anonymizedItems";
+    }
+
+    @Override
+    public String getDescription() {
+        return Messages.AnonymizedItems_Description();
+    }
+
+    public Map<String, String> getAnonymizedItemMap() {
+        return Anonymizer.getDisplayItems();
+    }
+
+    public boolean isAnonymizeLabels() {
+        return settings.isAnonymizeLabels();
+    }
+
+    public void setAnonymizeLabels(boolean anonymizeLabels) {
+        settings.setAnonymizeLabels(anonymizeLabels);
+    }
+
+    public boolean isAnonymizeItems() {
+        return settings.isAnonymizeItems();
+    }
+
+    public void setAnonymizeItems(boolean anonymizeItems) throws IOException {
+        settings.setAnonymizeItems(anonymizeItems);
+    }
+
+    public boolean isAnonymizeViews() {
+        return settings.isAnonymizeViews();
+    }
+
+    public void setAnonymizeViews(boolean anonymizeViews) throws IOException {
+        settings.setAnonymizeViews(anonymizeViews);
+    }
+
+    public boolean isAnonymizeNodes() {
+        return settings.isAnonymizeNodes();
+    }
+
+    public void setAnonymizeNodes(boolean anonymizeNodes) throws IOException {
+        settings.setAnonymizeNodes(anonymizeNodes);
+    }
+
+    public boolean isAnonymizeComputers() {
+        return settings.isAnonymizeComputers();
+    }
+
+    public void setAnonymizeComputers(boolean anonymizeComputers) throws IOException {
+        settings.setAnonymizeComputers(anonymizeComputers);
+    }
+
+    public boolean isAnonymizeUsers() {
+        return settings.isAnonymizeUsers();
+    }
+
+    public void setAnonymizeUsers(boolean anonymizeUsers) throws IOException {
+        settings.setAnonymizeUsers(anonymizeUsers);
+    }
+
+    @RequirePOST
+    public void doUpdateNow(StaplerRequest request, StaplerResponse response) throws ServletException, IOException {
+        Anonymizer.refresh();
```

Comment:
Is there an issue if this is called while a support bundle is being generated? I think it is fine as long as the config hasn't changed, but if it has changed would it be possible to end up with invalid mappings?

Reply:
I don't think so - I think that what would happen is there would be inconsistency within the support bundle.

That said, I honestly have a real lack of experience with dealing with multiple threads, so I'd greatly appreciate someone looking at the potential flaws in `Anonymizer` wrt multithreading.

Reply:
This could be addressed by making `Anonymizer::refresh` a `synchronized` method, though that may be overkill depending on where it's used.